server/handlers.go

```diff
@@ -138,13 +138,13 @@ func (s *Server) discoveryHandler() (http.HandlerFunc, error) {
 func (s *Server) handleAuthorization(w http.ResponseWriter, r *http.Request) {
 	authReq, err := s.parseAuthorizationRequest(s.supportedResponseTypes, r)
 	if err != nil {
-		s.renderError(w, http.StatusInternalServerError, err.Type, err.Description)
+		s.renderError(w, http.StatusInternalServerError, errServerError, "Failed to parse authorization request")
```

Comment:
A lot of the existing http errors actually mentioned "authorization request". I'll clean those up too then.